src/org/loklak/client/SearchClient.java

```diff
@@ -31,39 +31,95 @@
-        Timeline tl = new Timeline(order);
-        String urlstring = "";
-        urlstring = protocolhostportstub + "/api/search.json?q=" + URLEncoder.encode(query.replace(' ', '+'), "UTF-8") + "&timezoneOffset=" + timezoneOffset + "&maximumRecords=" + count + "&source=" + (source == null ? "all" : source) + "&minified=true&timeout=" + timeout;
-        JSONObject json = JsonIO.loadJson(urlstring);
-        if (json == null || json.length() == 0) return tl;
-        JSONArray statuses = json.getJSONArray("statuses");
+    private static final String SEARCH_API = "/api/search.json?q=";
+    private static final String ENCODING = "UTF-8";
+    // search API parameter fields
+    private static final String PARAM_TIMEZONE_OFFSET = "&timezoneOffset=";
+    private static final String PARAM_MAXIMUM_RECORDS = "&maximumRecords=";
+    private static final String PARAM_SOURCE = "&source=";
+    private static final String PARAM_MINIFIED = "&minified=";
+    private static final String PARAM_TIMEOUT = "&timeout=";
+    // search API parameter values
+    private static final String PARAM_SOURCE_VALUE = "all";
+    private static final String PARAM_MINIFIED_VALUE = "true";
+    // key of JSONObjects
+    private static final String JSON_KEY_STATUSES = "statuses";
+    private static final String JSON_KEY_USER = "user";
+    private static final String JSON_KEY_SEARCH_METADATA = "search_metadata";
+    private static final String JSON_KEY_HITS = "hits";
+    private static final String JSON_KEY_SCRAPER_INFO = "scraperInfo";
+
+    // restricts instantiation of SearchClient
+    private SearchClient() {}
+
+    // TODO: documentation of order
+    // TODO: use enumeration for source parameter (api users can't provide vague values)
+    /**
+     *Returns a {@link Timeline} object containing the search results.
+     * @param hostServerUrl Url of server hosting loklak_server
+     * @param query query string
+     * @param order
+     * @param source tweet searching source, possible values: cache, twitter, all (default)
+     * @param count maximum number of search result for the query
+     * @param timezoneOffset offset of a timezone in minutes
+     * @param timeout maximum waiting time of results in milliseconds
+     * @throws IOException
+     * @return Timeline object containing details and tweets of search result
+     */
+    public static Timeline search(final String hostServerUrl,
+                                  final String query,
+                                  final Timeline.Order order,
+                                  final String source,
+                                  final int count,
+                                  final int timezoneOffset,
+                                  final long timeout) throws IOException {
```

Comment:
This looks good. really long parameter methods can be broken down to have params on new lines. 👍